src/aws/osml/model_runner/inference/feature_utils.py

```diff
@@ -34,35 +31,41 @@ def features_to_image_shapes(sensor_model: SensorModel, features: List[Feature])
     :return: List[BaseGeometry] = a list of shapely shapes
     """
     shapes: List[BaseGeometry] = []
-    if not features:
+
+    if features is None:
+        logger.warning("Features cannot be None, returning empty list")
         return shapes
+
     for feature in features:
-        if "geometry" not in feature:
-            raise ValueError("Feature does not contain a valid geometry")
-
-        feature_geometry = feature["geometry"]
-
-        image_coords = convert_nested_coordinate_lists(feature_geometry["coordinates"], sensor_model.world_to_image)
-
-        feature_geometry["coordinates"] = image_coords
-
-        if isinstance(feature_geometry, Point):
-            shapes.append(shapely.geometry.Point(image_coords))
-        elif isinstance(feature_geometry, LineString):
-            shapes.append(shapely.geometry.LineString(image_coords))
-        elif isinstance(feature_geometry, Polygon):
-            shapes.append(shapely.geometry.shape(feature_geometry))
-        elif isinstance(feature_geometry, MultiPoint):
-            shapes.append(shapely.geometry.MultiPoint(image_coords))
-        elif isinstance(feature_geometry, MultiLineString):
-            shapes.append(shapely.geometry.MultiLineString(image_coords))
-        elif isinstance(feature_geometry, MultiPolygon):
-            shapes.append(shapely.geometry.shape(feature_geometry))
-        else:
-            # Unlikely to get here as we're handling all valid geojson types but if the spec
-            # ever changes or if a consumer passes in a custom dictionary that isn't valid
-            # we want to handle it gracefully
-            raise ValueError("Unable to convert feature due to unrecognized or invalid geometry")
+        try:
```

Comment:
Why was this function changed to skip geometries instead of raise exceptions?

Reply:
If there are valid features in the list do we not want to process them instead of dropping the entire thing and warn the user that a subset was missed?

Reply:
Updated to include logic to toggle the skipping of bad features.